dashboard/modules/serve/serve_head.py

```diff
@@ -14,81 +13,130 @@
 routes = dashboard_optional_utils.ClassMethodRouteTable
 
 
+class RestMethod(str, Enum):
+    GET = "GET"
+    PUT = "PUT"
+    DELETE = "DELETE"
+
+
 class ServeHead(dashboard_utils.DashboardHeadModule):
     def __init__(self, dashboard_head):
         super().__init__(dashboard_head)
         self._http_session = aiohttp.ClientSession()
+        self.rest_method_executors = {
+            RestMethod.GET: self._http_session.get,
+            RestMethod.PUT: self._http_session.put,
+            RestMethod.DELETE: self._http_session.delete,
+        }
 
-    @routes.get("/api/serve_head/applications/")
-    @dashboard_optional_utils.init_ray_and_catch_exceptions()
-    async def get_serve_instance_details(self, req: Request) -> Response:
-        """
-        This proxies to the serve_agent's /api/serve/applciations call
+    async def proxy_request(self, req: Request, route: str, method: RestMethod):
+        """Forwards the req request to the Serve agent on the head node.
+
+        Args:
+            req: request to forward.
+            route: route to send the request to.
+            rest_method: REST method to use when sending the request.
         """
-        head_agent_address = await self._get_head_agent()
-        if not head_agent_address:
-            return Response(
-                status=503,
-                text=(
-                    "Failed to find the serve agent. "
-                    "Check the dashboard_agent logs to see if the agent "
-                    "failed to launch."
-                ),
-            )
+
+        head_agent_address = "http://localhost:52365"
```

Comment:
is it intentional hardcoded here? 
nit: Use `DEFAULT_DASHBOARD_AGENT_LISTEN_PORT`

Reply:
I hardcoded this because there's a small propagation delay before the Serve head receives the dashboard port when a Ray cluster starts. I'll change it back to dynamic though because users may change their dashboard agent port. Thanks for the catch!

Reply:
It gets the address dynamically now.